File: C++/Count-Primes.cpp
class Solution {
public:
 
    int countPrimes(int n) {
 
        vector<bool> arr(n+1,0);
        arr[0]=true;
        arr[1]=true;
        for(int i=2;i*i<=n;i++){
            if(arr[i]==false){
                for(int j = i*i;j<=n;j+=i){
                    arr[j]=true;
                }
            }
        }
        
        int count = 0;
        for(int i=2;i<n;i++){
            if(!arr[i]){
                count++;
            }
        }
        return count;
    }
};

File: Python/count_primes.py
class Solution(object):
    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """
        isPrime = [True] * n
        for i in xrange(2, n):
            if i * i >= n:
                break
            if not isPrime[i]:
                continue
            for j in xrange(i * i, n, i):
                isPrime[j] = False
        count = 0
        for i in xrange(2, n):
            if isPrime[i]:
                count += 1
        return count
